aten/src/ATen/cpu/vec/vec512/vec512_bfloat16.h

```diff
@@ -1348,41 +1348,19 @@ static inline void _transpose_mxn_half_32_32(__m512i r[], __m512i d[]) {
 // Code referred to FBGEMM:
 // https://github.com/pytorch/FBGEMM/blob/39a423e4ad1a04b77fea81c7d09c3e6f8984fae9/src/UtilsAvx512.cc#LL19C6-L19C6
 template<>
-inline void transpose_mxn<BFloat16, 32, 32>(
-    const BFloat16* src,
-    int64_t ld_src,
-    BFloat16* dst,
-    int64_t ld_dst) {
-  // Load from memory
-  __m512i r[32];
-#ifndef __msvc_cl__
-#pragma unroll(32)
-#endif
-  for (int i = 0; i < 32; ++i) {
-    r[i] = _mm512_loadu_si512(reinterpret_cast<const __m512i*>(src + i* ld_src));
-  }
-
-  __m512i d[32];
-  _transpose_mxn_half_32_32(r, d);
-
-  // Store to dst
-#ifndef __msvc_cl__
-#pragma unroll(32)
-#endif
-  for (int i = 0; i < 32; ++i) {
-    _mm512_storeu_si512(dst + i* ld_dst, d[i]);
-  }
-}
-
-template <typename T, int M, int N,
-          typename std::enable_if_t<std::is_same<T, BFloat16>::value && ((M < 32 && M != 16) || (N < 32 && N != 16)), int> = 0>
-inline void transpose_mxn(const BFloat16* src, int64_t ld_src, BFloat16* dst, int64_t ld_dst) {
+inline void transpose_mxn<BFloat16>(const BFloat16* src, int64_t ld_src, BFloat16* dst, int64_t ld_dst, int M, int N) {
```

Comment:
I'm concerned about the perf impact. Originally, M and N are template args and compile-time constants.

Reply:
No performance regression in TAS, and we compared assembly code before and after PR, it doesn't seem to have  performance impact.

Reply:
Thanks! So perhaps we don't have to keep the version with `M` and `N` as template args.